### PR TITLE
Allow booting of solo5/ukvm with the boot command

### PR DESCRIPTION
**Pull request by niks3089**
Changes done:

- The IncludeOS can now be booted on solo5 as follows
`boot --with-solo5 .`
- Changed test_ukvm.sh to reflect this change.
- vm.json file is not required for solo5 currently but can be used if required in the future.
-  Created a new class ukvm under hypervisor
- Moved start_process() function to the parent class(hypervisor). Similarly, stop(), wait() and other functions could be moved to the parent class.

---
This change is [Reviewable](https://reviewable.io/reviews/hioa-cs/includeos/1706)

### CODE REVIEW

**Comment by AndreasAakesson** [comment]
Cool! 
ok to test

File: vmrunner/vmrunner.py [hunk 1]
```diff
@@ -180,6 +180,136 @@ def name(self):
...
+        self.info = Logger(color.INFO("<" + type(self).__name__ + ">"))
+
+    def name(self):
+        return "Ukvm"
+
+    def image_name(self):
+        return self._image_name
+
+    def drive_arg(self):
+        return ["--disk=" + INCLUDEOS_HOME + "dummy.disk"]
```
Comment (ricarkol) [inline vmrunner/vmrunner.py]: Some services need a real disk image, like https://github.com/hioa-cs/IncludeOS/tree/master/test/fs/integration/fat32.

Comment (niks3089) [reply vmrunner/vmrunner.py]: I see, I guess we can use vm.json to pass the disk and net parameters when not present defaults to dummy.disk and tap100. 
Also, instead of setting tap interface and creating dummy.disk in boot.py, we can move that to vmrunner.py where it can be configured based on vm.json

A sample vm.json
```
{
  "net" : "tap101"
  "disk"   : "another.disk"
}
``` 
Thoughts?

**Comment by fwsGonzo** [comment]
Thank you for contributing to IncludeOS! This is a much needed feature for our solo5 buffs.

It's hard to follow your commit messages, have you read our guidelines?
http://includeos.readthedocs.io/en/latest/Contributing-to-IncludeOS.html

The easiest fix would be to rebase everything into a single commit, something ala "boot: Add --with-solo5 option". Also, you can add the disk support in a separate commit later.

File: vmrunner/vmrunner.py (hunk 1 above)
Comment (ricarkol) [reply vmrunner/vmrunner.py]: Was thinking what to do about the TAPs. Qemu runs get a script that's used to create the TAP. In the case of solo5/ukvm, it seems that vmrunner should be the place where the TAPs are created (and the disk).

**Comment by niks3089** [comment]
@fwsGonzo : Re-based all the changes as a single commit for this PR as well for PR #1707
